packages/velaux-ui/src/pages/ApplicationConfig/components/TriggerList/index.tsx

```diff
@@ -79,31 +82,57 @@ class TriggerList extends Component<Props, State> {
     });
   };
 
+  handleCustomTriggerTab = (token: string) => {
+    this.setState({ customTriggerType: token });
+  }
+
   render() {
     const { Row, Col } = Grid;
     const { triggers, applicationDetail } = this.props;
 
-    const { showTrigger, component } = this.state;
+    const { showTrigger, component, customTriggerType } = this.state;
 
     const domain = `${window.location.protocol}//${window.location.host}`;
     const webHookURL = `${domain}/api/v1/webhook/${showTrigger?.token}`;
     let command = `curl -X POST -H 'content-type: application/json' --url ${webHookURL}`;
 
     if (showTrigger?.payloadType == 'custom' && component) {
-      const body = {
-        upgrade: {
-          [component.name]: {
-            image: component.properties && component.properties.image,
+      const customTriggerBody: {[x: string]: Object} = {
+        execute: {
+          action: 'execute',
+          upgrade: {
+            [component.name]: {
+              image: component.properties && component.properties.image,
+            },
+          },
+          codeInfo: {
+            commit: '',
+            branch: '',
+            user: '',
           },
         },
-        codeInfo: {
-          commit: '',
-          branch: '',
-          user: '',
+        approve:{
+          action: 'approve',
+          step : 'suspend'
         },
-      };
+        terminate:{
+          action: 'terminate',
+          step : 'suspend'
+        },
+        rollback :{
+          action: 'rollback',
+          step : 'suspend'
+        }
+      }
+      let body;
+      for (let key in customTriggerBody) {
+        if(key == customTriggerType) {
+          body = customTriggerBody[key];
+        }
+      }
```

Comment:
```suggestion
      let body = customTriggerTyoe[customTriggerBody];
```

Reply:
```
let body = customTriggerTyoe[customTriggerBody];
```

is throwing error because `customTriggerType` is undefined initially. So something like this will work ig.

```
let body =  customTriggerBody ? customTriggerBody[customTriggerType] : " ";
```

Reply:
done !